### PR TITLE
Mongoid 7 support

### PR DESCRIPTION
**Pull request by abhi8686**

### CODE REVIEW

File: lib/mongoid/embedded_errors/embedded_in.rb [hunk 1]
```diff
@@ -1,12 +1,8 @@
-if Gem::Version.new(Mongoid::VERSION) >= Gem::Version.new('7.0.0')
-  require 'mongoid/association/embedded/embedded_in'
-  ASSOCIATION = Mongoid::Association::Macros::ClassMethods
-  EMBEDS_MANY = Mongoid::Association::Embedded::EmbedsMany
-else
-  require 'mongoid/relations/embedded/in'
-  ASSOCIATION = Mongoid::Relations::Macros::ClassMethods
-  EMBEDS_MANY = Mongoid::Relations::Embedded::Many
-end
+# frozen_string_literal: true
```
Comment (knovoselic) [inline lib/mongoid/embedded_errors/embedded_in.rb:1]: Let's keep this compatible with both older and new Mongoid. There aren't any benefits in dropping support of the older version.

File: lib/mongoid/embedded_errors.rb [hunk 2]
```diff
@@ -6,31 +8,33 @@ module Mongoid::EmbeddedErrors
   def self.included(klass)
     return if klass.instance_methods.include?(:errors_without_embedded_errors)
 
-    klass.public_send :alias_method, :errors_without_embedded_errors, :errors
-    klass.public_send :alias_method, :errors, :errors_with_embedded_errors
+    klass.send :alias_method, :errors_without_embedded_errors, :errors
```
Comment (knovoselic) [inline lib/mongoid/embedded_errors.rb:11]: Why was this change introduced?

File: lib/mongoid/embedded_errors.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (knovoselic) [inline lib/mongoid/embedded_errors.rb:1]: Can you explain the changes in this file in general? I'm not sure how these are needed for compatibility with newest Mongoid.

Comment (abhi8686) [reply lib/mongoid/embedded_errors.rb:1]: i dont think we need it as of . there are no strings needed that needs to be frozen

File: lib/mongoid/embedded_errors.rb (hunk 2 above)
Comment (knovoselic) [reply lib/mongoid/embedded_errors.rb:9]: https://github.com/glooko/mongoid-embedded-errors/pull/18

File: lib/mongoid/embedded_errors/version.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
+# frozen_string_literal: true
+
 module Mongoid; end
 module Mongoid::EmbeddedErrors
-  VERSION = '2.0.1'.freeze
+  VERSION = '3.0.0'
```
Comment (knovoselic) [inline lib/mongoid/embedded_errors/version.rb:5]: Given that we're not breaking functionality, let's increase version to 2.1.0

File: lib/mongoid/embedded_errors.rb [hunk 5]
```diff
@@ -6,31 +6,33 @@ module Mongoid::EmbeddedErrors
...
+          next unless rel.errors.any?
+
           # get each of their individual message and add them to the parent's errors:
           rel.errors.each do |k, v|
-            relation = if Gem::Version.new(Mongoid::VERSION) >= Gem::Version.new('7.0.0')
-                         metadata.class
-                       else
-                         metadata.relation
-                       end
+            relation = metadata.class
```
Comment (knovoselic) [inline lib/mongoid/embedded_errors.rb:28]: Does this work with all versions of mongoid? 4 through 7?

File: lib/mongoid/embedded_errors.rb (hunk 3 above)
Comment (knovoselic) [reply lib/mongoid/embedded_errors.rb]: I've asked if you could explain changes in the file, all changes, in general. Why are they necessary, as this looks like a lot of unnecessary checks.

I didn't ask about the magic comment - it should definitely be added to all files.

**Review by knovoselic — CHANGES_REQUESTED** [review]

File: lib/mongoid/embedded_errors/version.rb (hunk 4 above)
Comment (knovoselic) [reply lib/mongoid/embedded_errors/version.rb]: Scratch this, we already have 2.1.x on rubygems somewhow. Let's bump this up to 3.0.0 as we're removing support for Mongoid 3.

File: lib/mongoid/embedded_errors.rb (hunk 3 above)
Comment (knovoselic) [reply lib/mongoid/embedded_errors.rb]: Ok, so all of the checks for empty arrays and so on are the changes from latest gemversion on rubygems?

File: lib/mongoid/embedded_errors.rb [hunk 6]
```diff
@@ -36,7 +38,7 @@ def errors_with_embedded_errors
                     "#{name}.#{k}"
                   end.to_sym
             errs.delete(key)
-            errs.add key, v if v.present?
+            errs.add key, v
```
Comment (knovoselic) [inline lib/mongoid/embedded_errors.rb]: I thought this `if v.present?` check is necessary, why was it removed?

Comment (abhi8686) [reply lib/mongoid/embedded_errors.rb]: its not needed. we are skipping if any errors.any? on line 26.

File: lib/mongoid/embedded_errors.rb (hunk 3 above)
Comment (abhi8686) [reply lib/mongoid/embedded_errors.rb]: yes i havent added any checked

**Comment by knovoselic** [comment]
@abhi8686 Can you please add the magic comment back in lib/mongoid/embedded_errors file? It shouldn't have been removed.

File: lib/mongoid/embedded_errors.rb (hunk 6 above)
Comment (knovoselic) [reply lib/mongoid/embedded_errors.rb]: I think this still might be needed in some edge cases where key is added to the errors array, but value is an empty string or something like that. Let's keep it for now, just to be safe.

**Review by knovoselic — APPROVED** [review]

**Pull request merged**